arch/x86/crc32_fold_pclmulqdq.c

```diff
@@ -384,18 +381,17 @@ static const unsigned ALIGNED_(16) crc_mask2[4] = {
     0x00000000, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF
 };
 
-uint32_t Z_INTERNAL crc_fold_512to32(unsigned int crc0[4 * 5]) {
+Z_INTERNAL uint32_t crc32_fold_final_pclmulqdq(crc32_fold *crc) {
     const __m128i xmm_mask  = _mm_load_si128((__m128i *)crc_mask);
     const __m128i xmm_mask2 = _mm_load_si128((__m128i *)crc_mask2);
 
-    uint32_t crc;
     __m128i x_tmp0, x_tmp1, x_tmp2, crc_fold;
 
     /* CRC_LOAD */
-    __m128i xmm_crc0 = _mm_loadu_si128((__m128i *)crc0 + 0);
-    __m128i xmm_crc1 = _mm_loadu_si128((__m128i *)crc0 + 1);
-    __m128i xmm_crc2 = _mm_loadu_si128((__m128i *)crc0 + 2);
-    __m128i xmm_crc3 = _mm_loadu_si128((__m128i *)crc0 + 3);
+    __m128i xmm_crc0 = _mm_loadu_si128((__m128i *)crc->fold + 0);
+    __m128i xmm_crc1 = _mm_loadu_si128((__m128i *)crc->fold + 1);
+    __m128i xmm_crc2 = _mm_loadu_si128((__m128i *)crc->fold + 2);
+    __m128i xmm_crc3 = _mm_loadu_si128((__m128i *)crc->fold + 3);
```

Comment:
Using temporary variable would make this cleaner as it would remove need for repeated casting.
Using unaligned version should also be unnecessary as the structure is guaranteed to be aligned and `fold` is the first member of the structure.

Reply:
If I remember correctly, there was a PR some time ago that converted everything to use relative paths. If you do a search for "../../" in code base there is a lot of instances.

Reply:
Fixed.